generals/rewards/reward_fn.py

```diff
@@ -0,0 +1,32 @@
+import abc
+
+from generals.core.action import Action
+from generals.core.observation import Observation
+
+
+class RewardFn(abc.ABC):
+    @abc.abstractmethod
+    def __call__(self, prior_obs: Observation, prior_action: Action, obs: Observation) -> float:
+        """
+        It's common to see notation like r(s, a) in RL theory and it's accordingly common for environments
```

Comment:
thats true (that it is very common), but i feel like in practice the post-action observation is even more important and i even see it in literature, so I think this is much better 👍

Reply:
Mmm cool! I've been reading the chapter on RL from https://udlbook.github.io/udlbook/. I know folks highly recommend Sutton & Barto -- is that what you're reading/read?

Reply:
Yeah, I've been reading that and it is a very cool book. I get most of my knowledge from our uni course tho. It is epic :D https://ufal.mff.cuni.cz/courses/npfl139/2324-summer

Reply:
Mmm nice. Looks good! I noticed you share a last name with the instructor -- are y'all related?